### PR TITLE
feat(provider): Add provider Calendly

### PR DESCRIPTION
**Pull request by ghassen-create**
# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: allauth/socialaccount/providers/calendly/views.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    access_token_url = "https://auth.calendly.com/oauth/token"
+    authorize_url = "https://auth.calendly.com/oauth/authorize"
+    profile_url = "https://api.calendly.com/users/me"
+
+    def complete_login(self, request, app, token, **kwargs):
+        resp = requests.get(
+            self.profile_url,
+            headers={"Authorization": f"BEARER {token.token}"},
+        )
+        print(resp.json())
```
Comment (pennersr) [inline allauth/socialaccount/providers/calendly/views.py:23]: **Issue**: :printer:

File: docs/overview.rst [hunk 2]
```diff
@@ -67,6 +67,8 @@ Supported Providers
 
 - Box (OAuth2)
 
+- CALENDLY (OAuth2)
```
Comment (pennersr) [inline docs/overview.rst:70]:
```suggestion
- Calendly (OAuth2)
```

File: docs/providers.rst [hunk 3]
```diff
@@ -304,6 +304,32 @@ Development callback URL
     http://localhost:8000/accounts/box/login/callback/
 
 
+CALENDLY
```
Comment (pennersr) [inline docs/providers.rst:307]:
```suggestion
Calendly
```

File: docs/providers.rst [hunk 4]
```diff
@@ -304,6 +304,32 @@ Development callback URL
...
+----
+App registration (get your key and secret here)
+    https://calendly.com/integrations/api_webhooks
+
+CALENDLY OAuth2 Documentation
```
Comment (pennersr) [inline docs/providers.rst:312]:
```suggestion
Calendly OAuth2 Documentation
```

**Review by pennersr — CHANGES_REQUESTED** [review]

**Comment by pennersr** [comment]
Can you have a look -- the test suite is failing.

**Comment by ghassen-create** [comment]
i am not able find an issue -- have a check now

**Pull request closed without merging**